### PR TITLE
[Samples][21.corebot-app-insights] Migrate DialogAndWelcomeBot class

### PR DESCRIPTION
**Pull request by FedericoBernal**
Fixes #

## Specific Changes

  - 
  -
  -

### CODE REVIEW

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.corebot.app.insights;
+
+import com.codepoetics.protonpack.collectors.CompletableFutures;
```
Comment (matiasroldan6) [inline samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: is this import all right?

Comment (Batta32) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: This is not ok

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import com.microsoft.bot.builder.MessageFactory;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.builder.UserState;
+import com.microsoft.bot.dialogs.Dialog;
+import com.microsoft.bot.schema.Activity;
+import com.microsoft.bot.schema.Attachment;
+import com.microsoft.bot.schema.ChannelAccount;
+import com.microsoft.bot.schema.Serialization;
+
+import java.awt.*;
```
Comment (Batta32) [inline samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:19]: Not sure about this

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.corebot.app.insights;
+
+import com.codepoetics.protonpack.collectors.CompletableFutures;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.io.IOUtils;
+import com.microsoft.applicationinsights.core.dependencies.apachecommons.lang3.StringUtils;
```
Comment (Batta32) [inline samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:8]: Is this ok? I mean, it's not ok to consume the dependencies of another dependency

**Review by matiasroldan6 — APPROVED** [review]
overall it all seems identical to sample 13 so I guess it's ok

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+            InputStream inputStream = Thread.currentThread().
+                getContextClassLoader().getResourceAsStream("cards/welcomeCard.json")
+        ) {
+            String adaptiveCardJson = IOUtils
+                .toString(inputStream, StandardCharsets.UTF_8.toString());
+
+            return new Attachment() {{
+                setContentType("application/vnd.microsoft.card.adaptive");
+                setContent(Serialization.jsonToTree(adaptiveCardJson));
+            }};
```
Comment (Batta32) [inline samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:92]: Be careful, DBI

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java (hunk 1 above)
Comment (FedericoBernal) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: It is identicall to the import in sample 13

**Review by Batta32 — CHANGES_REQUESTED** [review]
I guess that this file it should be a simply copy paste

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java (hunk 1 above)
Comment (Batta32) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: I know it's identical, but doesn't mean to be ok 😅

Comment (matiasroldan6) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: apparently it's used in line 74. Maybe we should review that

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java (hunk 2 above)
Comment (matiasroldan6) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:19]: nope, we shouldn't use wildcard imports, as per checkstyle

File: samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java (hunk 1 above)
Comment (Batta32) [reply samples/21.core-bot-app-insights/src/main/java/com/microsoft/bot/sample/corebot/app/insights/DialogAndWelcomeBot.java:6]: After checking, this is correct ,see another [sample](https://github.com/microsoft/botbuilder-java/blob/28fe834716f71b7c340e250bcd6e36dd53286ca9/samples/43.complex-dialog/src/main/java/com/microsoft/bot/sample/complexdialog/DialogAndWelcome.java#L6)

**Review by Batta32 — APPROVED** [review]

**Pull request merged**